src/Sarif/Visitors/PartitioningVisitor.cs

```diff
@@ -174,7 +174,7 @@ public override Result VisitResult(Result node)
 
                 Result partitionedResult = deepClone
                     ? node.DeepClone()
-                    : new Result(node);
```

Comment:
>new Result(node); [](start = 22, length = 17)

This code created a new Result reference but then shared properties from the original result (which could potentially be mutable). Instead, let's just reuse the result. This allows maximal reuse of the object. In our case, we update the shared result with the work item filing details and this therefore flows directly back to the original log file (even though we're filing against a partition of it).